ci/bin/format.dart

```diff
@@ -711,7 +705,7 @@ class WhitespaceFormatChecker extends FormatChecker {
     return true;
   }
 
-  static Future<_GrepResult> _hasTrailingWhitespace(File file) async {
+  static Future<_GrepResult?> _hasTrailingWhitespace(File file) async {
```

Comment:
What do you think about a `_GrepResult.empty(File file)` constructor that generates a result with empty hits/lineNumbers lists? Could be a separate patch if you prefer to focus primarily on the types here.